### PR TITLE
Create Dockerfile

### PR DESCRIPTION
**Pull request by ading1977**
This PR:

* Create Dockerfile
* Update Makefile
* Update `difTargetConfig` to `targetConfig` in the turbodif configuration
* Package `app-supply-chain-config.yaml` in the Docker container

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ENV GIT_COMMIT ${GIT_COMMIT}
+
+### Setup user for build execution and application runtime
+ENV APP_ROOT=/opt/turbonomic
+ENV PATH=$PATH:${APP_ROOT}/bin
+
+RUN mkdir -p ${APP_ROOT}/bin
+RUN mkdir -p ${APP_ROOT}/conf
+COPY build/turbodif.linux ${APP_ROOT}/bin/turbodif
+COPY configs/app-supply-chain-config.yaml ${APP_ROOT}/conf/app-supply-chain-config.yaml
```
Comment (esara) [inline build/Dockerfile:13]: do we want to externalize this into a configmap?

File: pkg/conf/conf.go [hunk 2]
```diff
@@ -13,10 +13,10 @@ import (
 const (
 	// Config file paths as volume mounts for the service pod
 	DefaultConfPath            = "/etc/turbodif/turbodif-config.json"
-	DefaultSupplyChainConfPath = "/etc/turbodif/app-supply-chain-config.yaml"
+	DefaultSupplyChainConfPath = "/opt/turbonomic/conf/app-supply-chain-config.yaml"
```
Comment (esara) [inline pkg/conf/conf.go:16]: why?

File: build/Dockerfile (hunk 1 above)
Comment (ading1977) [reply build/Dockerfile:13]: I originally externalize this into a `configmap`, then found out that a dynamic change to the supply chain config causes probe registration to fail on the Platform after a restart of the DIF probe:
```
topology-processor-5d74fc86cd-p6 }
topology-processor-5d74fc86cd-p6  is incompatible with already registered probe with the same probe type: probeType: "DataIngestionFramework"
```
On the platform side, we have a `ProbeInfoCompatibilityChecker` to compare certain probe fields for compatibility, so I feel that allow modifying of the supply chain config is risky. 

Plus updating the supply chain configmap needs update multiple places, including the deployment of every metric server (yamls, helm, operator), the deployment of the standalone DIF probe, and the deployment of XL prometurbo subchart.

File: pkg/conf/conf.go (hunk 2 above)
Comment (ading1977) [reply pkg/conf/conf.go:16]: Same reason above.

File: build/Dockerfile (hunk 1 above)
Comment (esara) [reply build/Dockerfile:13]: ok, dropping it

**Pull request merged**